tests/cases/compiler/elidedJSImport2.ts

```diff
@@ -0,0 +1,35 @@
+// @allowJs: true
+// @checkJs: false
+// @module: ES2022, commonjs
+// @moduleResolution: node16
+// @esModuleInterop: true
+// @isolatedModules: true

```

Comment:
I think those compiler options work ok with my current changes, but I am not sure. I also don't think any other compiler option could interact badly with this change, but I have limited knowledge of the many things that could go wrong with emit and imports, so please, let me know if there's something else I should be testing here.